src/opensles/AudioStreamBuffered.cpp

```diff
@@ -25,23 +25,18 @@ namespace oboe {
  * AudioStream with a FifoBuffer
  */
 AudioStreamBuffered::AudioStreamBuffered(const AudioStreamBuilder &builder)
-        : AudioStream(builder)
-        , mFifoBuffer(nullptr)
-{
+        : AudioStream(builder) {
+}
+AudioStreamBuffered::~AudioStreamBuffered() {
+    delete mFifoBuffer;
 }
 
-Result AudioStreamBuffered::open() {
-
-    Result result = AudioStream::open();
-    if (result != Result::OK) {
-        return result;
-    }
-
+Result AudioStreamBuffered::finishOpen() {
     // If the caller does not provide a callback use our own internal
     // callback that reads data from the FIFO.
     if (getCallback() == nullptr) {
-        LOGD("AudioStreamBuffered(): new FifoBuffer");
-        // TODO: Fix memory leak here
+        LOGD("AudioStreamBuffered(): new FifoBuffer(bytesPerFrame=%d", getBytesPerFrame());
+        // FIFO is configured with the same format and channels as the stream.
         mFifoBuffer = new FifoBuffer(getBytesPerFrame(), 1024); // TODO size?
         // Create a callback that reads from the FIFO
         mInternalCallback = std::unique_ptr<AudioStreamBufferedCallback>(new AudioStreamBufferedCallback(this));
```

Comment:
You can use std::unique_ptr::reset method http://en.cppreference.com/w/cpp/memory/unique_ptr/reset:

mInternalCallback.reset(new AudioStreamBufferedCallback(this))

Reply:
Thanks. I cleaned up this section quite a bit.